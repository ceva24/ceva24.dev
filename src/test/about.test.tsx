import React from "react";
import { render } from "@testing-library/react";
import About from "../pages/about";

jest.mock("gatsby", (): unknown => {
    return {
        ...jest.requireActual("gatsby"),
        graphql: jest.fn(),
        StaticQuery: jest.fn(),
        useStaticQuery: jest.fn(),
    };
});

jest.mock("@reach/router", () => ({
    useLocation: jest.fn().mockImplementation(() => {
        return { pathname: "/about/" };
    }),
}));

describe("about", () => {
    it("renders", () => {
        const { asFragment } = render(<About />);

        expect(asFragment()).toMatchSnapshot();
    });
});
