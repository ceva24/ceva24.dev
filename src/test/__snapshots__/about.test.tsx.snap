// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`about renders 1`] = `
<DocumentFragment>
  <div
    class="text-lg font-light max-w-4xl mx-auto"
  >
    <header
      class="border-b"
    >
      <div
        class="my-8 text-center space-y-5"
      >
        <h1
          class="text-6xl"
        >
          <a
            aria-current="page"
            class="text-black"
            href="/"
          >
            Chris Evans
          </a>
        </h1>
        <div
          class="px-6"
        >
          <strong>
            ceva24.dev
          </strong>
           | Thoughts, code and everything in-between
        </div>
        <div
          aria-label="Contact"
          class="space-x-8 text-2xl"
        >
          <a
            aria-label="Email"
            class="text-gray-800"
            href="mailto:chris@ceva24.dev"
            title="mailto:chris@ceva24.dev"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-envelope "
              data-icon="envelope"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M464 64C490.5 64 512 85.49 512 112C512 127.1 504.9 141.3 492.8 150.4L275.2 313.6C263.8 322.1 248.2 322.1 236.8 313.6L19.2 150.4C7.113 141.3 0 127.1 0 112C0 85.49 21.49 64 48 64H464zM217.6 339.2C240.4 356.3 271.6 356.3 294.4 339.2L512 176V384C512 419.3 483.3 448 448 448H64C28.65 448 0 419.3 0 384V176L217.6 339.2z"
                fill="currentColor"
              />
            </svg>
          </a>
          <a
            aria-label="Github"
            class="text-gray-800"
            href="https://www.github.com/ceva24"
            title="https://www.github.com/ceva24"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-github "
              data-icon="github"
              data-prefix="fab"
              focusable="false"
              role="img"
              viewBox="0 0 496 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3 .3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5 .3-6.2 2.3zm44.2-1.7c-2.9 .7-4.9 2.6-4.6 4.9 .3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3 .7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3 .3 2.9 2.3 3.9 1.6 1 3.6 .7 4.3-.7 .7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3 .7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3 .7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"
                fill="currentColor"
              />
            </svg>
          </a>
          <a
            aria-label="LinkedIn"
            class="text-gray-800"
            href="https://www.linkedin.com/in/ceva24"
            title="https://www.linkedin.com/in/ceva24"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-linkedin "
              data-icon="linkedin"
              data-prefix="fab"
              focusable="false"
              role="img"
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M416 32H31.9C14.3 32 0 46.5 0 64.3v383.4C0 465.5 14.3 480 31.9 480H416c17.6 0 32-14.5 32-32.3V64.3c0-17.8-14.4-32.3-32-32.3zM135.4 416H69V202.2h66.5V416zm-33.2-243c-21.3 0-38.5-17.3-38.5-38.5S80.9 96 102.2 96c21.2 0 38.5 17.3 38.5 38.5 0 21.3-17.2 38.5-38.5 38.5zm282.1 243h-66.4V312c0-24.8-.5-56.7-34.5-56.7-34.6 0-39.9 27-39.9 54.9V416h-66.4V202.2h63.7v29.2h.9c8.9-16.8 30.6-34.5 62.9-34.5 67.2 0 79.7 44.3 79.7 101.9V416z"
                fill="currentColor"
              />
            </svg>
          </a>
        </div>
      </div>
    </header>
    <nav
      class="my-6 space-x-12 min-w-max text-center"
    >
      <a
        aria-current="page"
        class="pb-2 text-black border-b border-transparent hover:border-black border-black"
        href="/"
      >
        POSTS
      </a>
      <a
        class="pb-2 text-black border-b border-transparent hover:border-black"
        href="/about/"
      >
        ABOUT
      </a>
    </nav>
    <main
      class="px-3 mb-6"
    >
      <h2>
        About me
      </h2>
      <p>
        Hi! I'm an Engineering Manager from the UK. I currently work at the LEGO Group, supporting a team of software engineers building great products and maintaining the systems that support the company's outstanding customer service.
      </p>
      <p>
        I started out in Java and then moved into back-end web development with Groovy/Grails. A while back I transitioned to JavaScript/TypeScript, Node.js and React (mainly Next.js and Gatsby). I've also built up a chunk of experience in AWS Serverless and event-driven architecture.
      </p>
      <p>
        I'm keen on automation and using it to increase teams' capabilities over time. I'm also interested in agile team leadership - driving change by focusing on user benefit, breaking things down into deliverable increments and using metrics to inform success. I value an open and collaborative approach to software development, always asking the question "what value are we delivering?"
      </p>
      <p>
        I'm passionate about people - attracting and recruiting talented developers, doing onboarding right, and supporting developers' careers to drive them towards success.
      </p>
      <p>
        I get satisfaction from seeing a team succeed and constantly improve itself, as well as knowing our work makes a difference to people.
      </p>
      <p>
        Please don't hesitate to 
        <a
          href="mailto:chris@ceva24.dev"
        >
          reach out to me
        </a>
         for a chat, to ask a question, or anything else really!
      </p>
      <h2>
        About this blog
      </h2>
      <p>
        I started this blog back in 2012 to have a bit more of a web presence, and to use it as a place to drop thoughts, code snippets, and anything else I wanted to talk about (hence the tagline "Thoughts, code and everything in-between"). It's fair to say I haven't made the most out of it, but I still like to update it every so often.
      </p>
      <p>
        Thanks for taking the time to check it out!
      </p>
    </main>
    <footer>
      <div
        class="min-w-max text-center"
      >
        <a
          aria-current="page"
          class=""
          href="/"
        >
          Home
        </a>
      </div>
      <div
        class="my-8 text-center space-y-3 pt-5 border-t max-w-lg mx-auto"
      >
        <div
          class="font-normal"
        >
          CHRIS EVANS
        </div>
        <div
          class="px-6"
        >
          Senior Engineering Manager at the LEGO Group, UK
        </div>
      </div>
    </footer>
  </div>
</DocumentFragment>
`;
